lib/src/instances/stream_monad.dart

```diff
@@ -68,10 +84,109 @@ class StreamMonad<T> extends Monad<T> implements Stream<T> {
   StreamMonad<E> asyncMap<E>(FutureOr convert(T event)) =>
       new StreamMonad(_stream.asyncMap(convert));
 
+  /// Combines multiple Observables to create an Observable whose values are
+  /// calculated from the latest values of each of its input Observables.
+  StreamMonad<E> combineLatest<S, E>(
+      StreamMonad<S> other, E combine(T element, S otherElement)) {
```

Comment:
Shouldn't be `combine` optional? of you provide it, it is used, if not, a tuple is created